src/SFA.DAS.EmploymentCheck.Functions/AzureFunctions/HouseKeeping/CleanupWorkflows_Timer.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+
+namespace SFA.DAS.EmploymentCheck.Functions.AzureFunctions.HouseKeeping
+{
+    public class CleanupWorkflows_Timer
+    {
+        [FunctionName(nameof(CleanupOldWorkflows))]
+        public async Task CleanupOldWorkflows([TimerTrigger("0 0 0 * * *"
+#if DEBUG
+                ,RunOnStartup = true
+#endif
+            )]
+            TimerInfo timerInfo,
+            [DurableClient] IDurableOrchestrationClient orchestrationClient, ILogger log)
+        {
+            var createdTimeFrom = DateTime.MinValue;
+            var createdTimeTo = DateTime.Now;
+            var runtimeStatus = new List<OrchestrationStatus>
+            {
+                OrchestrationStatus.Completed,
+                OrchestrationStatus.Canceled,
+                OrchestrationStatus.ContinuedAsNew,
+                OrchestrationStatus.Failed,
+                OrchestrationStatus.Terminated,
+            };
+            var result =
+                await orchestrationClient.PurgeInstanceHistoryAsync(createdTimeFrom, createdTimeTo, runtimeStatus);
+            log.LogInformation("Scheduled cleanup done, {InstancesDeleted} instances deleted", result.InstancesDeleted);
+        }
+    }
+}
```

Comment:
Is this timer running going to be running in pre-prod/prod? In which case could it run and clear 'state' while an Orchestrator is running?

Reply:
this is what is used in Employer Incentives. Let's assume we don't need this for now. Deleted.